third_party/cpgfunctionEP-0.3.2/CMakeLists.txt

```
@@ -0,0 +1,86 @@
+cmake_minimum_required(VERSION 3.6)
+project(cpgfunctionEP)
+
+set(CMAKE_CXX_STANDARD 17)
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --std=c++17")
+else()
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+    add_compile_definitions(_USE_MATH_DEFINES)
+endif()
+
+# Everything will need to access this
+include_directories(include)
+# Third party includes
+include_directories(third_party/eigen-3.3.9)  # Eigen for LU decomposition
+include_directories(third_party/qdt-master)
+
+# Create a DLL with an API
+add_library(cpgfunctionEP SHARED
+        src/boreholes.cpp
+        src/gfunction.cpp
+        src/heat_transfer.cpp
+        src/interpolation.cpp
+        src/utilities.cpp
+        src/coordinates.cpp
+        src/statistics.cpp
+        src/segments.cpp
+        src/blas.cpp)
+
+# It depends on threads
+find_package(Threads REQUIRED)
+target_link_libraries(cpgfunctionEP ${CMAKE_THREAD_LIBS_INIT})
+
+OPTION (USE_OpenMP "Use OpenMP" ON)
+IF(USE_OpenMP)
+    FIND_PACKAGE(OpenMP)
+    IF(OPENMP_FOUND)
+        SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
+        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+    ENDIF()
+ENDIF()
```

I am not a big fan of messing with CMAKE_C_FLAGS/CXX_FLAGS. The canonical way to do this is:

```
find_package(OpenMP)
if(OpenMP_CXX_FOUND)
    target_link_libraries(MyTarget PUBLIC OpenMP::OpenMP_CXX)
endif()
```

**I think `PRIVATE` linking is appropriate here, maybe it's `PUBLIC` though.**

```suggestion
option (USE_OpenMP "Use OpenMP" ON)
if(USE_OpenMP)
    find_package(OpenMP)
    if(OpenMP_CXX_FOUND)
      target_link_libraries(cpgfunctionEP PRIVATE OpenMP::OpenMP_CXX)
    else()
      message(AUTHOR_WARNING "OpenMP_CXX not found")
    endif()
endif()
```

---

Looks like private works.